src/CRM/CivixBundle/Resources/views/Code/module.civix.php.php

```diff
@@ -446,3 +446,26 @@ function _<?php echo $mainFile ?>_civix_civicrm_alterSettingsFolders(&$metaDataF
     $metaDataFolders[] = $settingsDir;
   }
 }
+
+/**
+ * (Delegated) Implements hook_civicrm_entityTypes().
+ *
+ * Find any *.entityType.php files, merge their content, and return.
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_entityTypes
+ */
+
+<?php
+// Add appropriate indentation
+foreach(explode("\n", $entityTypes) as $k => $l){
+  if($k){
+    $entityTypeLines[$k] = '  '.$l;
+  }else{
+    $entityTypeLines[$k] = $l;
+  }
+}
+$entityTypes = implode("\n", $entityTypeLines);
+?>
+function _<?php echo $mainFile ?>_civix_civicrm_entityTypes(&$entityTypes) {
+  $entityTypes += <?php echo $entityTypes ?>;
```

Comment:
I like the idea of aggregating the list ahead of time to avoid extra file-scans. 

This should be an `array_merge()` instead of `+` -- to my reading of `CRM_Utils_Hook::entityTypes(&$entityTypes)`, there's no specific key for `$entityTypes`, so we should assume numeric keys. For numeric keys, the `+` will cause values to [trample each other](https://gist.github.com/totten/22b2d64f1a11b1da28dc4fabf7ff39e0).

Reply:
Good point - that is now updated...